tvbwidgets/ui/connectivity_matrix_editor_widget.py

```diff
@@ -0,0 +1,233 @@
+import math
+import numpy as np
+import ipycanvas as canvas
+import ipywidgets as widgets
+import matplotlib.colors as mcolors
+from IPython.display import display
+from tvb.datatypes.connectivity import Connectivity
+
+class ConnectivityMatrixEditor():
+    def __init__(self, connectivity):
+        self.connectivity = connectivity
+        self.weights = self.connectivity.weights
+        self.tract_lengths = self.connectivity.tract_lengths
+        self.connectivities_history = [self.connectivity]
+        self.cell_size = 20
+        self.num_rows = int(len(self.connectivity.weights[0])/2)
+        self.num_cols = int(len(self.connectivity.weights[1])/2)  
+        self._make_header()
+        self.tab = widgets.Tab()
+        self._get_quadrant_range(selection = 1)
+        self._prepare_matrices_tab()
+        self.new_connectivity = self._prepare_new_connectivity()
+
+
+    def _make_header(self):
+        self.header = widgets.HBox()
+        options = ["Quadrant 1", "Quadrant 2", "Quadrant 3", "Quadrant 4"]
+        self.quadrants = widgets.Dropdown(options = options)
+        self.quadrants.observe(self._on_quadrant_select, names = ["value"])
+        self.cell_value = widgets.Text(description = "value")
+        self.button = widgets.Button(description = "Change")
+        self.button.on_click(lambda change :self.on_apply_change(change))
+        self.cell_value.layout.visibility = "hidden"
+        self.button.layout.visibility = "hidden"
```

Comment:
When you create a new widget you can add its layout directly in its definition. This approach is cleaner than adding it afterwards.
ex: self.button = widgets.Button(description = "Change", layout=widgets.Layout(width='80px', visibility='hidden'))